docs/guides/02-concepts/inter-process-communication.md

```diff
@@ -0,0 +1,58 @@
+---
+sidebar_position: 2
+---
+
+# Inter-Process Communication
+
+Inter-Process Communication (IPC) allows isolated processes to communicate securely and is key to building more complex applications.
+
+Tauri uses a particular style of Inter-Process Communication called [Asynchronous Message Passing], where processes exchange _requests_ and _responses_ serialized using some simple data representation. Message Passing should sound familiar to anyone with web development experience, as this paradigm is used for client-server communication on the internet.
+
+Message passing is a safer technique than shared memory or direct function access because the recipient is free to reject or discard requests as it sees fit. For example, if the Tauri Core process determines a request to be malicious, it simply discards the requests and never executes the corresponding function.
+
+In the following, we explain Tauri's two IPC primitives - `Events` and `Commands` - in more detail.
+
+## Events
+
+Events are fire-and-forget, one-way IPC messages that are best suited to communicate lifecycle events and state changes. Contrary to [Commands](#commands) Events can be emitted by both the Frontend _and_ the Tauri Core.
+
+<figure>
+
+```mermaid
+sequenceDiagram
+    participant F as Frontend
+    participant C as Tauri Core
+
+    C-)F: Event
+```
+
+<figcaption>Figure 1-2: An event sent from the Core to the Frontend.</figcaption>
+</figure>
+
+## Commands
+
+Tauri also provides a [foreign function interface]-like abstraction on top IPC messages[^1]. The primary API, `invoke`, is similar to the browsers `fetch` API and allows the Frontend to invoke rust functions, pass arguments, and receive data.
+
+Because this mechanism uses the [JSON-RPC] protocol under the hood to serialize requests and responses, all arguments and return data must be serializable to JSON.
```

Comment:
technically this is incorrect, isn't it? JSON-RPC is a specification and afaik we're not following that spec. I don't reallyyy have our ipc implementation in my head rn tho 🤔

Reply:
It's not a 100% compatible but it's very close to json-rpc.

Reply:
Let me check how json-rpc compliant we actually are!

Reply:
Alright I have no idea where that code ended up 🤷🏻‍♂️😅 I'll just reword it

Reply:
I adapted this wording now